src/DomainHost.sol

```diff
@@ -336,38 +368,47 @@ abstract contract DomainHost {
         emit Cage();
     }
 
-    /// @notice Set this domain's cure value
+    /// @notice Process this domain debt
     /// @param _lid Local ordering id
-    /// @param value The value of the cure [RAD]
-    function _tell(uint256 _lid, uint256 value) internal ordered(_lid) {
+    /// @param _debt The value of debt [RAD]
+    function _tell(uint256 _lid, uint256 _debt) internal ordered(_lid) {
         require(live == 0, "DomainHost/live");
-        require(!cureReported, "DomainHost/cure-reported");
-        require(_divup(value, RAY) <= grain, "DomainHost/cure-bad-value");
+        require(!debtReported, "DomainHost/debt-reported");
+
+        uint256 _grain = grain;
+        uint256 debtW = _divup(_debt, RAY);
+        if (_grain > debtW) {
+            uint256 wad = _grain - debtW + ddai;
```

Comment:
So `ddai` is different as whenever you have it is because there was already an action in L2 that removed it from there so we need to account that in case of GS. With the `dsin` is the opposite, you first act in L1, then in L2.
However there is something wrong there, as you mention we need to take into account `ddai` in the conditional but i also need to work on some extra scenarios as I think I might be missing some cases. I'll keep you updated as soon as I have news. cc @olivdb

Reply:
I applied the fix to `tell`, I'm still thinking if somehow that `transferFrom` can revert somehow, if there is a change that `grain - ddai - debtW` can be greater than the amount in the escrow. I think is not possible, as whatever makes to increase `ddai` is actual `debt` that's getting accounted. Will keep thinking but so far I've added an extra test scenario and reset the values of `existingEscrowDai` and `existingSurf` to zero to try to trigger some edge case.

Reply:
Fix looks good. I'll leave this comment open for the additional scenarios discussion.

Reply:
I kept thinking and I don't find a way for this to revert. If `debt` is correctly reported by the `Guest`, then it should never revert the `transferFrom`.

Reply:
Anyway as we stated this is an ongoing dev work, so we still need deeper reviews and we should probably incorporate FV specs and try to prove these invariants.

Reply:
Resolving this for now, if we identified some extra case we can work it in a separate issue/PR.